### ISSUE TITLE
[Boost] Click event handler on point is not called

### ISSUE THREAD

**Issue by burtyish** [highcharts/highcharts#14067]
#### Expected behaviour
When Boost mode is activated, a click handler provided in the options should be called when a point is clicked.

#### Actual behaviour
Click handler is not called when boost mode is active. 
When boost mode is not active, click handler works OK.

#### Live demo with steps to reproduce

Boosted chart with 3 points. Click handler on a point and see the handler isn't called.
https://codepen.io/burtyish/pen/XWdKMMV?editors=1011

If you raise the `boostThreshold` from 2 to 5 and click you will see the handler works as expected.

##### Note: 
I've opened this issue since #4569 was automatically closed.
I have yet to try out the workaround described there (using halo).

#### Product version
Highcharts Basic v8.1.2 

#### Affected browser(s)
Chrome

**Comment by burtyish** [comment]
If this is a limitation of the boost module, that should be specified along with other limitations in https://code.highcharts.com/modules/boost.js and other documentation

**Comment by pawelfus** [comment]
Thanks @burtyish! I think you are right, previous ticket was marked as an enhancement, so we should add this to other limitations when using boost.

PS:
Just copy&pasting the workaround from #4569:

Demo: https://jsfiddle.net/BlackLabel/4fka1xje/

Snippet:
```js
    plotOptions: {
        series: {
            point: {
                events: {
                    click: function () {
                        alert('a');
                    },
                    mouseOver: function() {
                        if(this.series.halo) {
                            this.series.halo.attr({
                               'class': 'highcharts-tracker' 
                            }).toFront();   
                        }
                    }
                }
            }
        }
    },
```

**Comment by burtyish** [comment]
@pawelfus yes thanks, the workaround works. 

Feel free to modify the issue title or open a new issue requiring a change to the documentation.

**Comment by pawelfus** [comment]
Marking as Undecided to be picked up next.

**Comment by Osokarp** [comment]
Any updates to this? Have logged an thread on the forums: https://www.highcharts.com/forum/viewtopic.php?f=9&t=46414 and confirmed as an issue, the workaround does not work on boost mode.

Thank you.

**Comment by KacperMadej** [comment]
A workaround was provided - https://jsfiddle.net/BlackLabel/4fka1xje/1/
If there are any problems with it please let us know.

This is a continuation of the previously identified issue, so it should be labeled similarly.

**Comment by Osokarp** [comment]
Hi @KacperMadej, it does not work for Bubble chart, and it is confirmed by Dominik at https://www.highcharts.com/forum/viewtopic.php?f=9&t=46414. Any alternative solutions? The halo attribute in the series will be undefined.

**Comment by KacperMadej** [comment]
Here is a workaround for bubble series: https://jsfiddle.net/BlackLabel/fs37Lka6/1/

**Comment by Osokarp** [comment]
Hi @KacperMadej, the workaround works fine for the bubble chart, however there's an error I face when I specify the series data as an array of objects instead of a 2d array. We need to do this as we need to attach some metadata to each point that we need to refer to later in the mouseover / click event. Tried the same thing on the jsfiddle you provided but couldn't replicate it.

I am using Highcharts v9.1.0 (according to the jsfiddle) and highcharts-react-official v3.0.0. Any ideas?

![image](https://user-images.githubusercontent.com/28076915/118210031-63f79580-b49c-11eb-96c5-14521b925b17.png)

**Comment by KacperMadej** [comment]
I move some code that needed Highcharts instance into a code block that could be [initialized as a module](https://github.com/highcharts/highcharts-react#how-to-add-a-module) in React environment (JS lines 14-36): https://jsfiddle.net/BlackLabel/fs37Lka6/2/

Working demo in React: https://stackblitz.com/edit/react-1db9sz?file=index.js

Boost mode requires data to be passed in the array format, so you could use [`keys`](https://api.highcharts.com/highcharts/series.bubble.keys) for additional data/metadata and still use the array format.

**Comment by Osokarp** [comment]
Hi @KacperMadej,

Thank you for your solution. It resolves the issue for Bubble chart. However, the same solution causes the click events for Scatter charts to not work (also in boost mode, hence using the same halo method). Is there a way to create another global Highcharts object that just has this additional feature which will just be used for bubble charts while leaving other charts unaffected?

Appreciate any other solutions or any best practices that I am not aware of?

**Comment by KacperMadej** [comment]
Updated the wrapper code to make needed changes only in case of bubble type series - https://jsfiddle.net/BlackLabel/fs37Lka6/3/

**Comment by HugoDD-PF** [comment]
> Updated the wrapper code to make needed changes only in case of bubble type series - https://jsfiddle.net/BlackLabel/fs37Lka6/3/

Hello, I tried adapting your solution for the heatmap type but it does not seems to work. Maybe the heatmap is implemented in a different way? Any help would be greatly appreciated!

**Comment by KacperMadej** [comment]
Implementation for the heatmap is different and unfortunately, there is no workaround for that series type that I know.

**Comment by AugustinMortier** [comment]
Thanks a lot for providing a workaround which allows clicking on markers when the boost mode is enabled. I would like to set the cursor as 'pointer' when hovering a marker to make it obvious that the click event is enabled. Did anybody managed to achieve that? The usual plotOptions does not seem to work in this case:
```
plotOptions: {
 series: {
  cursor: "pointer"
  }
}
```

**Comment by karolkolodziej** [comment]
Please add `cursor: pointer;` to the `highcharts-tracker` class as shown in [the demo.](https://jsfiddle.net/BlackLabel/sn5wg4cf/)

**Comment by AugustinMortier** [comment]
Great! I didn't think about the css style option. Thanks a lot @karolkolodziej !

**Comment by hortizgarrido** [comment]
I'm stuck as well. The workaround does not work for heatmap in boost mode. Any updates on this?

**Comment by karolkolodziej** [comment]
Unfortunately, I also haven't found any workaround for the `heatmap`.

**Comment by markzolotoy** [comment]
01/8/2025. HC version 11.2
Line chart. Point click does not fire. Tried under series and line, same thing

**Comment by hubertkozik** [comment]
@markzolotoy This issue was fixed in Highcharts v11.4.2: https://jsfiddle.net/BlackLabel/q6uao2e7/
